bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,875 @@
+                localWebsocketClient.connect(this, uri, request);
+
+                logger.debug("Websocket connection to SysAP is OK, timeout: {}", BRIDGE_WEBSOCKET_TIMEOUT);
+
+                ret = true;
+            } else {
+                ret = false;
+            }
+        } catch (Exception e) {
+            logger.debug("Error by opening Websocket connection [{}]", e.getMessage());
+
+            if (localWebsocketClient != null) {
+                try {
+                    localWebsocketClient.stop();
+
+                    ret = false;
+                } catch (Exception e1) {
+                    logger.debug("Error by opening Websocket connection [{}]", e1.getMessage());
+
+                    ret = false;
+                }
+            } else {
+                ret = false;
+            }
+        }
+
+        return ret;
+    }
+
+    /**
+     * Method to close the websocket connection
+     */
+    public void closeWebSocketConnection() {
+
+        socketMonitor.interrupt();
+
+        QueuedThreadPool localThreadPool = jettyThreadPool;
+
+        if (localThreadPool != null) {
+            try {
+                localThreadPool.stop();
+            } catch (Exception e1) {
+                logger.debug("Error by closing Websocket connection [{}]", e1.getMessage());
+            }
+        }
+
+        WebSocketClient localWebSocketClient = websocketClient;
+
+        if (localWebSocketClient != null) {
+            try {
+                localWebSocketClient.stop();
+            } catch (Exception e2) {
+                logger.debug("Error by closing Websocket connection [{}]", e2.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Method to open the websocket connection
+     */
+    public boolean openWebSocketConnection() {
+        boolean ret = false;
+
+        QueuedThreadPool localThreadPool = jettyThreadPool;
+
+        if (localThreadPool == null) {
+            jettyThreadPool = new QueuedThreadPool();
+
+            localThreadPool = jettyThreadPool;
+
+            if (localThreadPool != null) {
+                localThreadPool.setName(FreeAtHomeBridgeHandler.class.getSimpleName());
+                localThreadPool.setDaemon(true);
+                localThreadPool.setStopTimeout(0);
+
+                ret = true;
+            }
+        }
+
+        WebSocketClient localWebSocketClient = websocketClient;
+
+        if (localWebSocketClient == null) {
+            websocketClient = new WebSocketClient(httpClient);
+
+            localWebSocketClient = websocketClient;
+
+            if (localWebSocketClient != null) {
+                localWebSocketClient.setExecutor(jettyThreadPool);
+
+                socketMonitor.start();
+
+                ret = true;
+            } else {
+                ret = false;
+            }
+        } else {
+            ret = true;
+        }
+
+        return ret;
+    }
+
+    /**
+     * Method to initialize the bridge
+     *
+     *
```

Comment:
```suggestion
```

Reply:
**Fixed**

Change to configuration. Therefore the adding via config file shall be possible. However not yet tested